flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/WorkingDirectory.java

```diff
@@ -29,20 +29,32 @@
  */
 public class WorkingDirectory {
     private final File root;
+    private final File tmp;
 
     private WorkingDirectory(File root) throws IOException {
         this.root = root;
+        createDirectory(root);
 
-        if (!root.mkdirs() && !root.exists()) {
+        this.tmp = new File(root, "tmp");
+        createDirectory(tmp);
+        FileUtils.cleanDirectory(tmp);
```

Comment:
Are there any safeguards that the WorkingDirectory is not shared across (active) processes?

Reply:
No, this is something what the user has to guarantee if he configures a working directory.